packages/plugin-essentials/sources/commands/set/version.ts

```diff
@@ -94,10 +99,39 @@ export default class SetVersionCommand extends BaseCommand {
   }
 }
 
-export async function findVersion(configuration: Configuration, request: `stable` | `canary`) {
-  const data = await httpUtils.get(`https://repo.yarnpkg.com/tags`, {configuration, jsonResponse: true});
+export async function resolveRange(configuration: Configuration, request: string) {
+  const range = semverUtils.validRange(request)!;
+
+  const data: Tags = await httpUtils.get(`https://repo.yarnpkg.com/tags`, {configuration, jsonResponse: true});
+
+  const candidates = data.tags.filter(version => {
+    try {
+      const candidate = new semverUtils.SemVer(version);
+
+      // Ignore prereleases
+      if (candidate.prerelease.length > 0)
+        return false;
+
+      if (!range.test(candidate))
+        return false;
+
+      return true;
+    } catch { }
```

Comment:
Why do we need a `try/catch`?

Reply:
We don't, thanks for noticing! I copied this code from `NpmSemverResolver` where the try/catch is used to ignore invalid semver. In our case, all of our versions are valid semver so we don't need this.